hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -866,8 +874,50 @@ protected void initializeZKBasedSystemTrackers()
 
   // Will be overriden in test to inject customized AssignmentManager
   @VisibleForTesting
-  protected AssignmentManager createAssignmentManager(MasterServices master) {
-    return new AssignmentManager(master);
+  protected AssignmentManager createAssignmentManager(MasterServices master,
+    LocalStore localStore) {
+    return new AssignmentManager(master, localStore);
```

Comment:
Good.

Was going to suggest AM ask the Master for its localStore but that probably TMI for the AM to know of. This is better separation of concerns.

Good.